src/serializer.cpp

```diff
@@ -138,59 +138,57 @@ prepare() ->
 
     if(st_ == style::source)
     {
-        if(! is_chunked_)
+        if(more_)
```

Comment:
Btw, what does this branch do here?

I removed it locally and tests still passed so this is either extraneous or the tests aren't comprehensive enough.

Reply:
we check `more_`  to see if there is any need to call `source::read`.
`more` is set with the following condition:
```C++
more_ = ! rv.finished;
```
If we call `source::read` while it has previously returned the finished flag, it leads to writing additional zero-length chunks every time `serializer::prepare` is called.

Reply:
So what would it take to make a failing test case if the initial `if(more_)` isn't there?

Reply:
Btw, I've found that if add this to the `test_source`:
```cpp
        results
        on_read(
            buffers::mutable_buffer b)
        {
            BOOST_TEST(!s_.empty());
```
this seems sufficient. This way we ensure that `on_read()` is never called once the serializer is empty.

Reply:
Yeah, that's a good idea.